tests/js/reactions.js

```diff
@@ -181,6 +246,72 @@ suite('Custom Element Reactions', function() {
 
   });
 
+  suite('customized built-in elements', function() {
+    test('inherits native behavior such as a button\'s disabled property', function(done) {
+      var numClickCalls = 0;
+      var isDisabled;
+
+      class CustomizedBuiltin7 extends HTMLButtonElement {
+        constructor() {
+          super();
+          this.disabled = true;
+          isDisabled = true;
+        }
+      }
+
+      customElements.define('customized-builtin-7', CustomizedBuiltin7, {extends: 'button'});
+      var button = document.createElement('button', {is: 'customized-builtin-7'});
+      var shouldBeDisabled = true;
+
+      button.addEventListener('click', function() {
+        if (isDisabled) {
+          assert.fail();
+          done();
+        } else {
+          done();
+        }
+      });
+      button.click();
+
+      setTimeout(function() {
+        isDisabled = false;
+        button.disabled = false;
+        button.click();
+      });
+    });
+
+    test('does not upgrade elements that don\'t match what extends was at definition', function() {
```

Comment:
This is something I am actually not 100% sure on in the spec. It seems like you should not be able to `<div is="x-foo" />` if you say `customElements.define('x-foo', XFoo, {extends: 'span'})`. But I was not able to find a definitive answer in the spec. Am happy to change if I'm wrong on this.